File: 3336-water-bottles-ii/water-bottles-ii.cpp
class Solution {
public:
    int maxBottlesDrunk(int numBottles, int numExchange) {
        int cnt = 0;
        while(numBottles >= numExchange){
            cnt += numExchange;
            numBottles -= numExchange;
            numExchange += 1;
            numBottles += 1;
        }
        return cnt + numBottles;
    }
};

File: 3336-water-bottles-ii/water-bottles-ii.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        cnt = 0
        while numBottles >= numExchange:
            cnt += numExchange
            numBottles -= numExchange
            numBottles += 1
            numExchange += 1
        return cnt + numBottles